File: AJChallengeBucketList/app/src/main/java/com/example/ajchallengebucketlist/MainActivity.java
package com.example.ajchallengebucketlist;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;
import androidx.cardview.widget.CardView;

public class MainActivity extends AppCompatActivity {

    CardView placesWantToGoCardView;
    CardView thinsToDoCardView;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        placesWantToGoCardView = findViewById(R.id.card_view_places_want_to_go);
        thinsToDoCardView = findViewById(R.id.card_view_things_to_do);

        placesWantToGoCardView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent placesIntent = new Intent(MainActivity.this, PlacesActivity.class);
                startActivity(placesIntent);
            }
        });

        thinsToDoCardView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent thingsIntent = new Intent(MainActivity.this, ThingsActivity.class);
                startActivity(thingsIntent);
            }
        });
    }
}

File: AndroidBuildFormChallange/app/src/main/java/com/example/buildformchallange/MainActivity.kt
package com.example.buildformchallange

import android.R.layout.simple_spinner_dropdown_item
import android.annotation.SuppressLint
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.ArrayAdapter
import com.example.buildformchallange.databinding.ActivityMainBinding

@SuppressLint("StaticFieldLeak")
private lateinit var binding: ActivityMainBinding

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        setUpSpinner()
        setUpButton()
    }

    private fun setUpButton() {
        val titles = arrayOf(
            "Miss", "Mrs", "Ms", "Mr", "Mx", "Dr", "Prof"
        )

        val titleAdapter = ArrayAdapter<String>(this, simple_spinner_dropdown_item, titles)
        binding.spinnerTitle.adapter = titleAdapter
    }

    private fun setUpSpinner() {
        binding.buttonCreateAccount.setOnClickListener {
            createAccount()
        }
    }

    private fun createAccount() {
        val user = User(
            binding.spinnerTitle.selectedItem as String,
            binding.editTextFirstName.text.toString(),
            binding.editTextLastName.text.toString(),
            binding.editTextEmail.text.toString(),
            binding.editTextPhoneNumber.text.toString(),
            binding.editTextPassword.text.toString(),
        )

        val summaryActivityIntent = Intent(this, SummaryActivity::class.java)
        summaryActivityIntent.putExtra("user_information", user)

        startActivity(summaryActivityIntent)

    }
}